### PR TITLE
🔎 Search fixes

### PR DESCRIPTION
**Pull request by rowanseymour**
* Add support for `uuid = ` and `uuid != `
* Add support for `urn = ` and `urn != ` and `urn ~ ` for non-anon orgs
* Allow checking whether URNs are set even when anon, e.g. `urn != ""` or `tel = ""`
* Don't localize search error messages which are no longer seen

### CODE REVIEW

File: locale/en_US/LC_MESSAGES/django.po [hunk 1]
```diff
@@ -2355,69 +2355,6 @@ msgstr ""
 msgid "The search query"
 msgstr ""
 
-#, python-brace-format
-msgid "Unrecognized field: '{prop}'"
-msgstr ""
-
-#, python-brace-format
-msgid "f'{val}' isn't a valid number"
```
Comment (rowanseymour) [inline locale/en_US/LC_MESSAGES/django.po:2363]: kudos to the translators who were translating this 🤦

File: temba/contacts/models.py [hunk 2]
```diff
@@ -797,20 +797,15 @@ def as_json(self):
...
-        obj["name"] = self.name
-        obj["id"] = self.id
-
-        return obj
+        return {
+            "id": self.id,
+            "uuid": self.uuid,
+            "name": self.name,
+            "language": self.language,
+            "urns": [dict(scheme=urn.scheme, path=urn.path) for urn in self.urns.all()],
```
Comment (rowanseymour) [inline temba/contacts/models.py:805]: always return urns so we can check existence even in the anon case

File: temba/contacts/search/parser.py [hunk 3]
```diff
@@ -70,8 +69,6 @@ def get_prop_map(self, org, validate=True):
         prop_map = {p: None for p in all_props}
 
         all_contact_fields = ContactField.all_fields.filter(org=org, key__in=all_props, is_active=True)
-        if not org.is_anon:
-            all_contact_fields.exclude(key="id")
```
Comment (rowanseymour) [inline temba/contacts/search/parser.py:74]: not allowed in dynamic group queries anyway

File: temba/contacts/models.py (hunk 2 above)
Comment (nicpottier) [reply temba/contacts/models.py:805]: Should we mask the path here just to be extra sure we don't leak somewhere?

File: locale/en_US/LC_MESSAGES/django.po (hunk 1 above)
Comment (nicpottier) [reply locale/en_US/LC_MESSAGES/django.po:2363]: Don't these get shown to the user in the case of making a query with an invalid syntax?

File: temba/contacts/search/parser.py [hunk 4]
```diff
@@ -501,10 +539,16 @@ def evaluate(self, org, contact_json, prop_map):
                     else:
                         return True
 
+            elif field_key == "urn":
+                if is_set:
+                    return contact_json["urns"]
+                else:
+                    return not contact_json["urns"]
```
Comment (nicpottier) [inline temba/contacts/search/parser.py:546]: Minor but maybe worth returning true `True` / `False` due to anonymous case.

File: locale/en_US/LC_MESSAGES/django.po (hunk 1 above)
Comment (rowanseymour) [reply locale/en_US/LC_MESSAGES/django.po:2363]: that's all coming from mailroom's endpoint now

**Pull request merged**